sentry/src/main/java/io/sentry/util/TracingUtils.java

```diff
@@ -0,0 +1,112 @@
+package io.sentry.util;
+
+import io.sentry.Baggage;
+import io.sentry.BaggageHeader;
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import io.sentry.PropagationContext;
+import io.sentry.Scope;
+import io.sentry.SentryOptions;
+import io.sentry.SentryTraceHeader;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class TracingUtils {
+
+  public static void startNewTrace(final @NotNull IHub hub) {
+    hub.configureScope(
+        scope -> {
+          scope.setPropagationContext(new PropagationContext());
+        });
+  }
+
+  public static @Nullable TracingHeaders traceIfAllowed(
+      final @NotNull IHub hub,
+      final @NotNull String requestUrl,
+      @Nullable List<String> thirdPartyBaggageHeaders,
+      final @Nullable ISpan span) {
+    final @NotNull SentryOptions sentryOptions = hub.getOptions();
+    if (sentryOptions.isTraceSampling() && shouldAttachTracingHeaders(requestUrl, sentryOptions)) {
+      return trace(hub, thirdPartyBaggageHeaders, span);
+    }
+
+    return null;
+  }
+
+  public static @Nullable TracingHeaders trace(
+      final @NotNull IHub hub,
+      @Nullable List<String> thirdPartyBaggageHeaders,
+      final @Nullable ISpan span) {
+    final @NotNull SentryOptions sentryOptions = hub.getOptions();
+
+    if (span != null && !span.isNoOp()) {
+      return new TracingHeaders(
+          span.toSentryTrace(), span.toBaggageHeader(thirdPartyBaggageHeaders));
+    } else {
+      final @NotNull TracingHeadersHolder returnValue = new TracingHeadersHolder();
```

Comment:
H: This is not thread-safe, any other thread that calls `configureScope` and mutates the `propagationContext`, or the `baggage` simultaneously will race.
I still recommend doing something similar to the `Session`.
`Session withSession(final @NotNull IWithSession sessionCallback)` that has its own lock and is able to perform multiple operations atomically.
@markushi what do you think?

Reply:
Hmm trying to think through where we'd have a race here. For backend requests there should be a separate hub and scope per request.

Reply:
Android is single hub/scope.

Reply:
I'll update the code.

Reply:
Feels like the underlying issue is `hub.configureScope` / the scope callback. I guess it makes sense to not allow two callbacks to be executed at the same time. I can follow up with an extra ticket for this if you agree.

Reply:
@markushi you mean synchronizing `configureScope` in general? Not sure about potential deadlocks due to other locks not combining well with this.

Reply:
I mean synchronizing the execution of the scope callback
Something like ~
````
public void configureScope(final @NotNull ScopeCallback callback) {
    final Scope scope = stack.peek().getScope();
    // maybe have a protected scope.updateLock?
    synchronized (scope) {
        callback.run(scope);
    }
}
````
But yeah, I'm not 100% either sure 😄

Reply:
Yeah that's what I had in mind. Just pushed pretty much that but only for `TracingUtils`. We may be able to change locking for the next major but it's probably a pain for users in case they suddenly start getting deadlocks. Please let me know what you think about the latest commit.

Reply:
The current `synchronized` approach locks the whole `scope` causing contention in case more threads wanna update something else on the scope.
The approach similar to `Session withSession(final @NotNull IWithSession sessionCallback)` solves this problem, is there any reason why to not just go this way? this approach is being used for Sessions and is working well.